File: cases/taint/java/context-pair-depth2-negative/ContextPairDepth2Negative.java
package dataflowbench.taint;

final class ContextPairDepth2Negative {
    static int dfb_source() { // DFB-SOURCE: context-pair-depth2-input
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: context-pair-depth2-sink

    static int helper(int value) { // DFB-WITNESS: context-pair-depth2-helper
        return value;
    }

    static int wrapper(int value) { // DFB-WITNESS: context-pair-depth2-wrapper
        return helper(value);
    }

    static int outerTainted() {
        return wrapper(dfb_source());
    }

    static int outerClean() {
        return wrapper(0);
    }

    static void run() {
        int tainted = outerTainted();
        int clean = outerClean();
        dfb_sink(clean);
    }
}

File: cases/taint/csharp/context-pair-depth2-negative/ContextPairDepth2Negative.cs
namespace DataFlowBench;

static class ContextPairDepth2Negative
{
    static string dfb_source() // DFB-SOURCE: context-pair-depth2-input
    {
        return "tainted";
    }

    static void dfb_sink(string value) { } // DFB-SINK: context-pair-depth2-sink

    static string Helper(string value) // DFB-WITNESS: context-pair-depth2-helper
    {
        return value;
    }

    static string Wrapper(string value)
    {
        return Helper(value);
    }

    static string OuterTainted()
    {
        return Wrapper(dfb_source());
    }

    static string OuterClean()
    {
        return Wrapper("clean");
    }

    static void Run()
    {
        string tainted = OuterTainted();
        string clean = OuterClean();
        dfb_sink(clean);
    }
}
